src/main/java/com/example/CinemaRoom/service/SeatsService.java

```diff
@@ -1,53 +1,62 @@
 package com.example.CinemaRoom.service;
 
+import com.example.CinemaRoom.dto.SeatResponse;
+import com.example.CinemaRoom.dto.SeatsResponse;
 import com.example.CinemaRoom.exception.PurchaseException;
 import com.example.CinemaRoom.model.Seat;
-import java.util.ArrayList;
+import com.example.CinemaRoom.repository.SeatRepository;
+import lombok.Getter;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
 import java.util.List;
+import java.util.stream.Collectors;
 
+@Getter
+@Service
 public class SeatsService {
 
+    private SeatRepository seatRepository;
     private static final int ROWS = 9;
     private static final int COLUMNS = 9;
     private static final int PREMIUM_ROWS = 4;
     private static final int PREMIUM_PRICE = 10;
     private static final int STANDARD_PRICE = 8;
-    private final List<Seat> seats = new ArrayList<>();
 
-    public SeatsService() {
-        for (int row = 1; row <= ROWS; row++) {
-            for (int column = 1; column <= COLUMNS; column++) {
-                seats.add(new Seat(row, column, getPriceForRow(row)));
-            }
-        }
+    @Autowired
+    public void CinemaRoomApplication(SeatRepository seatRepository) {
```

Comment:
Shouldnt this be a constructor?

```suggestion
    public SeatsService(SeatRepository seatRepository) {
```